src/relax/transform/lift_transform_params.cc

```diff
@@ -43,26 +43,92 @@ struct LiftTransformParamsInfoPlan {
       lifted_bindings;  // the bindings of the original function that are lifted
 };
 
+namespace {
+template <typename T>
+struct InsertionOrderSet {
```

Comment:
I implemented something similar here https://github.com/apache/tvm/blob/unity/src/support/ordered_set.h

Reply:
Thank you, and I'll switch this PR over to it.

Though, related to it, `std::list` is usually implemented as a linked list, and has very poor performance as a result.  Is it currently being used in a context where the less-frequent iterator invalidation of `std::list` is required?

Reply:
That’s a good point. I agree vector is better when deletion is infrequent.

Reply:
And the PR is now updated to use `support::OrderedSet`.  I did need to make a few adjustments to it, so that it could support both hashable types (using `std::unordered_map<T, ValueType>`) and TVM objects (using `std::unordered_map<T, ValueType, ObjectPtrHash, ObjectPtrEqual>`), but that seemed like good functionality to have overall.